### DISCUSSION TITLE
Customization of Platform specific code

### DISCUSSION THREAD

**Discussion by daveytran** [korlibs/korge#868] [Q&A]
Is there documentation or recommended way on how to customise the generated Android code in my project? As it is, it just gets overwritten when i run a build

**Comment by soywiz** [comment]
This is the code for the `korge { }` block in your `build.gradle` for the android indirect mode (the default mode that creates a separate project for the android target):

https://github.com/korlibs/korge/blob/fe4eb1e1ad9eaec35b7ff79ae5ca434d53897826/buildSrc/src/main/kotlin/com/soywiz/korge/gradle/KorgeExtension.kt#L333

You can add extra code to gradle with this (including `apply from: '../../../your_custom_build.gradle'`):
https://github.com/korlibs/korge/blob/fe4eb1e1ad9eaec35b7ff79ae5ca434d53897826/buildSrc/src/main/kotlin/com/soywiz/korge/gradle/KorgeExtension.kt#L363

And then there are several methods to adjust how the manifest XML is generated:
https://github.com/korlibs/korge/blob/fe4eb1e1ad9eaec35b7ff79ae5ca434d53897826/buildSrc/src/main/kotlin/com/soywiz/korge/gradle/KorgeExtension.kt#L401-L420
